qutebrowser/browser/webkit/webkittab.py

```diff
@@ -50,6 +54,25 @@ def save_page(self):
         """Save the current page."""
         raise browsertab.UnsupportedOperationError
 
+    def show_source(self, dispatcher, url):
+        def show_source_cb(source):
+            """Show source as soon as it's ready."""
+            # WORKAROUND for https://github.com/PyCQA/pylint/issues/491
+            # pylint: disable=no-member
+            lexer = pygments.lexers.HtmlLexer()
+            formatter = pygments.formatters.HtmlFormatter(
+                full=True, linenos='table',
+                title='view-source:{}'.format(url.toDisplayString()))
+            # pylint: enable=no-member
+            highlighted = pygments.highlight(source, lexer, formatter)
+
+            new_tab = dispatcher._tabbed_browser.tabopen()
+            new_tab.set_html(highlighted)
+            new_tab.data.viewing_source = True
+            new_tab.url = lambda requested=False: QUrl(
```

Comment:
You're [monkey patching](https://en.wikipedia.org/wiki/Monkey_patch) the `url` method away here. Apart from this being something you should never do outside of test code, it's also broken: No matter what you navigate this tab to in the future, it'll always show this URL, as you patched the original code away.

Instead, the `set_html` method can take a second argument which is the URL. Please use that instead.